script.js

```diff
@@ -1,4 +1,250 @@
+  } // end of click submit message
+  else if (message == "result") {
+    if (equals == false) {
+      outputString = "Player " + playerNumber + ", you chose Dice " + input;
+    }
+
+    outputString =
+      outputString + "<br>" + "Your number is " + combinedNumber + "👍.";
+    if (playerNumber == 2) {
+      outputString =
+        outputString +
+        "<br>" +
+        winningPlayer +
+        " wins this round. Congrats!!🤝";
+    }
+    if (playerNumber == 1) {
+      outputString = outputString + " It is now Player 2's turn.";
+    }
+  } // end of result message
+  return outputString;
+};
+//function for first mode logic
+var rollDiceMode = function (input) {
+  if (input == "") {
+    randomNumberOne = rollADice();
+    randomNumberTwo = rollADice();
+
+    if (randomNumberOne == randomNumberTwo) {
+      equals = true;
+      combinedNumber = "" + randomNumberOne + randomNumberTwo;
+      if (playerNumber == 1) {
+        currentPlayerOneScore = Number(combinedNumber);
+        // playerOneScores.push(currentPlayerOneScore);
+        totalPlayerOneScore = totalPlayerOneScore + currentPlayerOneScore;
+      } else if (playerNumber == 2) {
+        currentPlayerTwoScore = Number(combinedNumber);
+        // playerTwoScores.push(currentPlayerTwoScore);
+        totalPlayerTwoScore = totalPlayerTwoScore + currentPlayerTwoScore;
+        if (currentPlayerOneScore > currentPlayerTwoScore) {
+          winningPlayer = "Player 1";
+        } else if (currentPlayerTwoScore > currentPlayerOneScore) {
+          winningPlayer = "Player 2";
+        }
+      }
+
+      myOutputValue = formatString(
+        "welcome",
+        input,
+        combinedNumber,
+        totalPlayerOneScore,
+        totalPlayerTwoScore
+      );
+      myOutputValue =
+        myOutputValue +
+        formatString(
+          "result",
+          input,
+          combinedNumber,
+          totalPlayerOneScore,
+          totalPlayerTwoScore
+        );
+      myOutputValue =
+        myOutputValue +
+        formatString(
+          "leaderboard",
+          input,
+          combinedNumber,
+          totalPlayerOneScore,
+          totalPlayerTwoScore
+        );
+
+      if (playerNumber == 1) {
+        playerNumber = 2;
+      } else if (playerNumber == 2) {
+        playerNumber = 1;
+      }
+    } else {
+      equals = false;
+
+      myOutputValue = formatString(
+        "welcome",
+        input,
+        combinedNumber,
+        totalPlayerOneScore,
+        totalPlayerTwoScore
+      );
+      myOutputValue =
+        myOutputValue +
+        formatString(
+          "leaderboard",
+          input,
+          combinedNumber,
+          totalPlayerOneScore,
+          totalPlayerTwoScore
+        );
+
+      gameMode = "resultMode";
+    }
+  } else {
+    myOutputValue = formatString(
+      "click submit",
+      input,
+      totalPlayerOneScore,
+      totalPlayerTwoScore
+    );
```

Comment:
Although the function `formatString()` takes in 5 arguments you don't need to fill in all of them if they are not going to be used in the function. In this case, if the first argument (`message`) is `"click submit"`, `formatString()` does not use any of the other arguments in its output. So, you can shorten this to `myOutputValue = formatString("click submit")` and it will still work the same!